### PR TITLE
Bugfix/gh782 acl xml format

### PR DESCRIPTION
**Pull request by kuenishi**
The root cause for #782 was misuse of `xmerl_scan:string/2`, which doesn't emit spaces, comments between actual xml elements. It leaves them as `#xmlText{}` and `#xmlComment{}` - This pull request commit changes to ignore them at walking around parsed dom tree.
This diffset also includes a small refactoring that wraps all `xmerl_scan:string/2` to one `riak_cs_xml` funcsion to control its usage.

### CODE REVIEW

**Comment by kuenishi** [comment]
Or maybe I'd open this against 1.4 branch?

**Comment by shino** [comment]
Agree. > 1.4

**Comment by reiddraper** [comment]
Yes, let's try and get this into 1.4.5.

File: src/riak_cs_xml.erl [hunk 1]
```diff
@@ -42,10 +44,34 @@
 -type internal_node() :: {atom(), [internal_node() | external_node()]} |
                          {atom(), attributes(), [internal_node() | external_node()]}.
 
+%% these types should be defined in xmerl.hrl or xmerl.erl
+%% for now they're defined here for convenience.
+-type xmlElement() :: #xmlElement{}.
+-type xmlText() :: #xmlText{}.
```
Comment (reiddraper) [inline src/riak_cs_xml.erl:50]: `xmlElement` and `xmlText` should be exported types (thanks dialyzer)

File: src/riak_cs_acl_utils.erl [hunk 2]
```diff
@@ -471,35 +473,40 @@ name_for_canonical(CanonicalId, RiakPid) ->
...
     case UpdAclRes of
         {ok, UpdAcl} ->
             process_acl_contents(RestElements, UpdAcl, RiakPid);
         {error, _}=Error ->
             Error
-    end.
+    end;
+process_acl_contents([#xmlComment{} | RestElements], Acl, RiakPid) ->
+    process_acl_contents(RestElements, Acl, RiakPid);
+process_acl_contents([#xmlText{value=" "} | RestElements], Acl, RiakPid) ->
```
Comment (reiddraper) [inline src/riak_cs_acl_utils.erl:504]: should we be skipping newlines too somewhere? Or is it only spaces that are causing trouble?

Comment (kuenishi) [inline src/riak_cs_acl_utils.erl:504]: This is what `{space, normalize}` does. Every space-like characters (newlines and tabs) are folded into single space " " by `erl_scan:string/2`. So we care only single space.

**Comment by kuenishi** [comment]
Reopened an issue against release/1.4 => #795.

File: src/riak_cs_xml.erl (hunk 1 above)
Comment (kuenishi) [inline src/riak_cs_xml.erl:50]: This is fixed in #795

File: src/riak_cs_acl_utils.erl (hunk 2 above)
Comment (reiddraper) [inline src/riak_cs_acl_utils.erl:504]: Cool, thanks.

**Pull request closed without merging**